### PR TITLE
Consolidated "Platforms Supported" and "More Platforms Tested" 

### PR DESCRIPTION
**Pull request by lrknox**
sections of RELEASE.txt into "Platforms Tested".

### CODE REVIEW

**Comment by lrknox** [comment]
I removed the Solaris and Windows 10 (32 bit) lines from the "Tested Configuration Features Summary" table.  There may be further additions or consolidations to that table.

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -669,40 +668,105 @@ Bug Fixes since HDF5-1.10.7 release
...
+    Linux 2.6.32-573.22.1.el6        GNU C (gcc), Fortran (gfortran), C++ (g++)
+    #1 SMP x86_64 GNU/Linux          compilers:
+    Centos6                              Version 4.4.7 20120313
+    (platypus)                           Version 4.9.3, 5.3.0, 6.2.0
+                                     MPICH 3.1.4 compiled with GCC 4.9.3
+                                     PGI C, Fortran, C++ for 64-bit target on
+                                     x86_64;
+                                         Version 19.10-0
+
+    Windows 10 x64                  Visual Studio 2015 w/ Intel Fortran 18 (cmake)
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt]: 2015 and 2017 also build with all Intel compilers.

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -669,40 +668,105 @@ Bug Fixes since HDF5-1.10.7 release
...
+                                    Visual Studio 2017 w/ Intel Fortran 19 (cmake)
+                                    Visual Studio 2019 w/ Intel Fortran 19 (cmake)
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt:799]: this one 2019 also builds with "Clang 12.0.0 with MSVC-like command-line" and no fortran.

**Comment by derobins** [comment]
I feel like it would be useful to have a paragraph that describes what should work and what we know doesn't work. For example, we know that certain versions of Windows won't work due to missing C99 support. I'm not sure if that sort of thing belongs here or elsewhere, though.

**Review by byrnHDF — COMMENTED** [review]
> I feel like it would be useful to have a paragraph that describes what should work and what we know doesn't work. For example, we know that certain versions of Windows won't work due to missing C99 support. I'm not sure if that sort of thing belongs here or elsewhere, though.

But our minimum is VS2015 and I think that is fine?

File: release_docs/RELEASE.txt [hunk 3]
```diff
@@ -669,40 +668,105 @@ Bug Fixes since HDF5-1.10.7 release
...
-
-    Windows 10 x64                Visual Studio 2015 w/ Intel Fortran 18 (cmake)
-                                  Visual Studio 2017 w/ Intel Fortran 19 (cmake)
-                                  Visual Studio 2019 w/ Intel Fortran 19 (cmake)
-                                  Visual Studio 2019 w/ MSMPI 10.1 (cmake)
-
-    macOS Mojave 10.14.6          Apple LLVM version 10.0.1 (clang-1001.0.46.4)
-    64-bit                        gfortran GNU Fortran (GCC) 6.3.0
-    (swallow)                     Intel icc/icpc/ifort version 19.0.4.233 20190416
+    Linux 5.10.10-200.fc33.x86_64    GNU gcc (GCC) 10.2.1 20201125 (Red Hat 10.2.1-9)
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt]: fedora34: 5.13.14-200.fc34
gcc: gcc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)
clang: clang version 12.0.1 (Fedora 12.0.1-1.fc34)

File: release_docs/RELEASE.txt [hunk 4]
```diff
@@ -669,40 +668,105 @@ Bug Fixes since HDF5-1.10.7 release
...
+    #1 SMP x86_64  GNU/Linux         GNU Fortran (GCC) 10.2.1 20201125 (Red Hat 10.2.1-9)
+    Fedora33                         clang version 11.0.0 (Fedora 11.0.0-2.fc33)
+                                     (cmake and autotools)
+
+    Linux 5.8.0-41-generic-x86_64    GNU gcc (GCC) 10.2.0-13ubuntu1
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt]: ubuntu 20.10: 5.8.0-63-generic #71-Ubuntu SMP
            gcc: gcc (Ubuntu 10.3.0-1ubuntu1~20.10) 10.3.0
            clang: Ubuntu clang version 11.0.0-2

ubuntu 20.04: 5.11.0-34-generic #36-20.04.1-Ubuntu SMP
            gcc: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
            clang: clang version 10.0.0-4ubuntu1

File: release_docs/RELEASE.txt [hunk 5]
```diff
@@ -669,40 +668,105 @@ Bug Fixes since HDF5-1.10.7 release
...
+    (platypus)                           Version 4.9.3, 5.3.0, 6.2.0
+                                     MPICH 3.1.4 compiled with GCC 4.9.3
+                                     PGI C, Fortran, C++ for 64-bit target on
+                                     x86_64;
+                                         Version 19.10-0
+
+    Windows 10 x64                  Visual Studio 2015 w/ Intel Fortran 18 (cmake)
+                                    Visual Studio 2017 w/ Intel Fortran 19 (cmake)
+                                    Visual Studio 2019 w/ Intel Fortran 19 (cmake)
+                                    Visual Studio 2019 w/ MSMPI 10.1 (cmake)
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt]: this MPI does no fortran

**Comment by byrnHDF** [comment]
BB defaults and StdShar config:
CMake
`      BUILD_SHARED_LIBS:
        BOOL: ON
      HDF5_BUILD_JAVA:
        BOOL: ON
      HDF5_BUILD_FORTRAN:
        BOOL: ON
      HDF5_BUILD_CPP_LIB:
        BOOL: ON
              HDF5_BUILD_DOC:
                BOOL: ON
`
autotools
`    fortran: enable
    cxx: enable
    java: enable
            doxygen: enable
`

**Comment by byrnHDF** [comment]
BB configs
StdStat:
CMake `      BUILD_SHARED_LIBS:
        BOOL: OFF
      HDF5_BUILD_JAVA:
        BOOL: OFF
`
autotools: `            static: enable
            shared: disable
            java: disable
`

**Comment by byrnHDF** [comment]
Other significant configs:
TSShar (StdShar with threadsafe)
CMake: `              HDF5_ENABLE_THREADSAFE:
                BOOL: ON
              ALLOW_UNSUPPORTED:
                BOOL: ON
`
autotools: `            threadsafe: enable
            unsupported: enable
`

StdMPIShar (StdShar with MPI)
CMake: `              HDF5_ENABLE_PARALLEL:
                 BOOL: ON
              HDF5_BUILD_CPP_LIB:
                 BOOL: OFF
              HDF5_BUILD_JAVA:
                 BOOL: OFF
`
autotools: `            parallel: enable
            cxx: disable
            java: disable
`
StdClang (CMake sanitizers)
Address: `              ONLY_SHARED_LIBS:
                BOOL: ON
              HDF5_BUILD_JAVA:
                BOOL: OFF
              HDF5_BUILD_FORTRAN:
                BOOL: OFF
              HDF5_ENABLE_SANITIZERS:
                BOOL: ON
              USE_SANITIZER:
                STRING: Address
              CMAKE_TOOLCHAIN_FILE:
                STRING: 'config/toolchain/clang.cmake'
`
Leak: `              ONLY_SHARED_LIBS:
                BOOL: ON
              HDF5_BUILD_JAVA:
                BOOL: OFF
              HDF5_BUILD_FORTRAN:
                BOOL: OFF
              HDF5_ENABLE_SANITIZERS:
                BOOL: ON
              USE_SANITIZER:
                STRING: Leak
              CMAKE_TOOLCHAIN_FILE:
                STRING: 'config/toolchain/clang.cmake'
`
Memory: `              ONLY_SHARED_LIBS:
                BOOL: ON
              HDF5_BUILD_JAVA:
                BOOL: OFF
              HDF5_BUILD_FORTRAN:
                BOOL: OFF
              HDF5_ENABLE_SANITIZERS:
                BOOL: ON
              USE_SANITIZER:
                STRING: Memory
              CMAKE_TOOLCHAIN_FILE:
                STRING: 'config/toolchain/clang.cmake'
`
Undefined: `              ONLY_SHARED_LIBS:
                BOOL: ON
              HDF5_BUILD_JAVA:
                BOOL: OFF
              HDF5_BUILD_FORTRAN:
                BOOL: OFF
              HDF5_ENABLE_SANITIZERS:
                BOOL: ON
              USE_SANITIZER:
                STRING: Undefined
              CMAKE_TOOLCHAIN_FILE:
                STRING: 'config/toolchain/clang.cmake'
`

**Review by epourmal — APPROVED** [review]

**Pull request closed without merging**